### PR TITLE
LPS-173333 Uses ClayEmptyState instead of adding custom logic

### PR DESCRIPTION
**Pull request by veroglez**
To review this pull I leave some tips on where each empty state is located 😉 

- **SelectAssetCategoryInfoItem**: Navigation Menu > Create a new NM > Add Category (the empty state should appear if there are no categories)
- **SelectCategory**: Content & Data > Web Content > Create a new WC > Categorization > Topic > Select button (the empty state should appear if there are no categories)
- **ChangeTrackingChangesView**: Applications Menu > Publications > Create a new Publication > Review Changes 
- **ChangeTrackingComments**: Applications Menu > Publications > Create a new Publication > Review Changes > Click to Comments button (the empty state should appear if there are no comments)
- **ChangeTrackingChangesView**: Applications Menu > Publications > Create a new Publication > Go to a page and edit it > Publish it > Go back to the previous Publication > Review Changes > Search and non-existent publication (the empty state should appear if there are no publications)
- **ChangeTrackingIndicator**: Applications Menu > Publications > Top publications selector > Select a Publication
-  **ChangeTrackingRenderView**: Applications Menu > Publications > Create a new Publication > Review Changes > Click on a change > Go to Parent or Children tab (the empty state should appear if there are no elements)
- **TreeFilter**: Applications Menu > Content Dashboard > Go to the second container (Content) > Click on Filter and Order > Select Extension… (the empty state should appear if there are no results)
- **EmptyState**: Forms > Create a new Form > Edit the Form > Go to Entries tab (the empty state should appear if there are no entries)
- **RuleList**: Forms > Create a new Form > Edit the Form > Go to Rules tab (the empty state should appear if there are no rules)
- **EmptyState (data-engine-taglib)**: Forms > Create a new Form > Edit the Form > Open Builder panel (right side) > Open Element Sets tab (the empty state should appear if there are no elements)
- **EmptyState (site-navigation)**: Site Builder > Navigation Menus > Add a Navigation Menu > Edit it (the empty state should appear if there are no elements)
- **SelectLayout**: Create a new Site > Content & Data > Web Content > Create a new Structure with a Link to Page > Create a WC with that structure > Edit it > Click on the Select of Link to Page input
- **Empty State (digital-signature)**: Applications Menu > Control Panel > Instance Settings > Digital Signature > Enable > Go to Content & Data > Digital Signature 
- **EmptyState (dxp/portal-workflow)**: Applications Menu > Workflow > Metrics > Search something or click in one metric > Workload by Assignee (right panel)
- **View.jsp (portal-search-web)**: Go to the Home > Search something in Search Bar > Search Result Portlet

**IMPORTANT**: There is also an open issue in Clay to fix the space between the image and the description (only when there is no title) https://github.com/liferay/clay/issues/5325

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by veroglez** [comment]
ci:test:sf

**Comment by ealonso** [comment]
@veroglez

**Comment by veroglez** [comment]
ci:test:sf

**Comment by ealonso** [comment]
Hey @veroglez,

I think we can also remove:

1. redirect-web/main.scss

```
.taglib-empty-result-message {
	margin-top: 5rem;
}
```

2. document-library-web/main.scss

```
.taglib-empty-result-message {
	background-color: $primary-l3;
	border-color: transparent;
}
```

3. dynamic-data-mapping-data-provider-web/main.scss

```
.taglib-empty-result-message {
	margin-top: 104px;
}
```

Also it should be really nice if we replace **liferay-ui:empty-result-message** with **liferay-frontend:empty-result-message**, it can be done in another task.

**Comment by veroglez** [comment]
Hi @ealonso! I have checked the scss that you comment and it seems that they are not used, so I have added a commit deleting it. About what you say about the **liferay-ui:empty-result-message**, I'm going to create a new technical task for that.

File: modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/js/components/table/EmptyState.js [hunk 1]
```diff
@@ -30,29 +30,22 @@ const DEFAULT_EMPTY = {
...
+			className={className}
+			description={description}
+			imgSrc={
+				className === DEFAULT_EMPTY.search.className
+					? `${themeDisplay.getPathThemeImages()}/states/search_state.gif`
+					: `${themeDisplay.getPathThemeImages()}/states/empty_state.gif`
+			}
+			title={title}
+		>
+			{button && button()}
```
Comment (victorg1991) [inline modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/js/components/table/EmptyState.js:47]: `button()` 🫠

**Comment by victorg1991** [comment]
ci:forward

**Comment by victorg1991** [comment]
Nice @veroglez 👏

**Comment by Tim-Cao** [comment]
See https://github.com/liferay-echo/liferay-portal/pull/10980

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**